### DISCUSSION TITLE
Vite server is not started

### DISCUSSION THREAD

**Discussion by asbjornu** [Eptagone/Vite.AspNetCore#119] [Q&A]
I have the following lines of code in my `Program.cs`:

```cs
builder.Services.AddViteServices();

// …

if (app.Environment.IsDevelopment())
{
    app.UseWebSockets();
    app.UseViteDevelopmentServer(true);
}
```

When starting the application, I get the following logged to the console:

```
warn: Vite.AspNetCore.ViteDevServerMiddleware[10]
      Connection refused (localhost:5173). Make sure the Vite development server is running
warn: Microsoft.AspNetCore.HttpsPolicy.HttpsRedirectionMiddleware[3]
      Failed to determine the https port for redirect.
info: Vite.AspNetCore.ViteManifest[1]
      The manifest file won't be read because the vite development service is enabled. The service will always return null chunks
```

From this, my understanding is that `npx vite` (or something similar) should be called somewhere, somehow, to launch the Vite development server. That, unfortunately, does not seem to happen. If I execute `npx vite` manually in the terminal, everything seems to start working, but my understanding was that `Vite.AspNetCore` was going to take care of that for me?

If it is by design that Vite needs to be started manually, I think it needs to be documented. If it is supposed to be started automatically, I'm in need of assistance, because it isn't happening. 🙏🏼

**Comment by Eptagone** [comment]
Hi @asbjornu. Did you set the `Server.AutoRun` setting to `true`?
You can see the [list of options](https://github.com/Eptagone/Vite.AspNetCore?tab=readme-ov-file#available-options) here.
Pass that option to the `AddViteServices` function, an user secret or you appsettings files.

**Reply by asbjornu** [reply]
Thanks! I now added this:

```cs
builder.Services.AddViteServices(options =>
{
    options.Server.AutoRun = true;
});
```

…but I still have to manually start Vite on the command line for things to start working.

**Reply by Eptagone** [reply]
Do you have your package.json in a subfolder instead of the project root? If that's the case, then set the `Server:PackageDirectory` option too.

**Reply by asbjornu** [reply]
`package.json` is in the same folder as the `.csproj`:

<img width="369" alt="Directory listing from Visual Studio Code showing package.json alongside Web.csproj in the same folder" src="https://github.com/Eptagone/Vite.AspNetCore/assets/12283/b1e0bf6d-3a37-4de5-973a-147b9ae4c976">

**Reply by Eptagone** [reply]
Which OS and package version are you using?

**Reply by asbjornu** [reply]
macOS Ventura 13.6.7, package version 2.1.0.

**Comment by asbjornu** [comment]
I'm also experiencing that scripts with the `vite-src` attribute are completely removed from the published Razor views. What am I doing wrong here?

**Reply by Eptagone** [reply]
Do you have a reproduction repo?
